### PR TITLE
Add 7 new vulnerbailities

### PR DESCRIPTION
**Pull request by vdeturckheim**

### CODE REVIEW

File: vuln/npm/405.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "id": 405,
+  "title": "Cross-site Scripting (XSS) - Stored",
+  "overview": "public allows to embed HTML in file names, which (in certain conditions) might lead to execute malicious JavaScript.",
+  "created_at": "2018-04-15",
+  "updated_at": "2018-04-15",
+  "publish_date": "2018-04-15",
+  "author": "bl4de (https://twitter.com/_bl4de)",
+  "module_name": "public",
+  "cves": [],
+  "vulnerable_versions": "",
```
Comment (lirantal) [inline vuln/npm/405.json]: let's use the latest version as vulnerable and the top limit, 
i.e: <=0.1.3

File: vuln/npm/406.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  "id": 406,
+  "title": "Cross-site Scripting (XSS) - Stored",
+  "overview": "There is a Stored XSS vulnerability in glance module. File name, which contains malicious HTML (eg. embedded iframe element or javascript: pseudoprotocol handler in <a> element) allows to execute JavaScript code against any user who opens directory listing contains such crafted file name.",
+  "created_at": "2018-04-15",
+  "updated_at": "2018-04-15",
+  "publish_date": "2018-04-15",
+  "author": "bl4de (https://twitter.com/_bl4de)",
+  "module_name": "glance",
+  "cves": [],
+  "vulnerable_versions": "",
```
Comment (lirantal) [inline vuln/npm/406.json]: here too

File: vuln/npm/408.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  "id": 408,
+  "title": "deep-extend prototype pollution",
+  "overview": "deep-extend node module suffers from a prototype pollution vulnerability",
+  "created_at": "2018-02-17",
+  "updated_at": "2018-04-09",
+  "publish_date": "2018-04-09",
+  "author": "Olivier Arteau (HoLyVieR)",
+  "module_name": "deep-extend",
+  "cves": [],
+  "vulnerable_versions": "",
```
Comment (lirantal) [inline vuln/npm/408.json]: also

File: vuln/npm/407.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  "id": 407,
+  "title": "deap prototype pollution",
+  "overview": "deap node module before 1.0.1 suffers from a prototype pollution vulnerability",
+  "created_at": "2018-02-17",
+  "updated_at": "2018-04-09",
+  "publish_date": "2018-04-09",
+  "author": "Olivier Arteau (HoLyVieR)",
+  "module_name": "deap",
+  "cves": [],
+  "vulnerable_versions": "<1.0.1",
```
Comment (lirantal) [inline vuln/npm/407.json]: <= , no?

File: vuln/npm/409.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  "id": 409,
+  "title": "merge-recursive prototype pollution",
+  "overview": "merge-recursive node module suffers from a prototype pollution vulnerability",
+  "created_at": "2018-02-17",
+  "updated_at": "2018-04-09",
+  "publish_date": "2018-04-09",
+  "author": "Olivier Arteau (HoLyVieR)",
+  "module_name": "merge-recursive",
+  "cves": [],
+  "vulnerable_versions": "",
```
Comment (lirantal) [inline vuln/npm/409.json]: also

File: vuln/npm/409.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  "patched_versions": "",
+  "slug": "merge-recursive-prototype-pollution",
+  "recommendation": "No fix is currently available for this vulnerability.\n\nIt is our recommendation to not install or use this module at this time.",
+  "references": "- https://hackerone.com/reports/311337",
```
Comment (lirantal) [inline vuln/npm/409.json]: report id is `311337`
am I the only one who finds it funny :-p

File: vuln/npm/410.json [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  "id": 410,
+  "title": "merge-options prototype pollution",
+  "overview": "merge-options node module suffers from a prototype pollution vulnerability",
+  "created_at": "2018-02-17",
+  "updated_at": "2018-04-09",
+  "publish_date": "2018-04-09",
+  "author": "Olivier Arteau (HoLyVieR)",
+  "module_name": "merge-options",
+  "cves": [],
+  "vulnerable_versions": "",
```
Comment (lirantal) [inline vuln/npm/410.json]: missing top version here

File: vuln/npm/411.json [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  "id": 411,
+  "title": "merge-objects prototype pollution",
+  "overview": "merge-objects node module suffers from a prototype pollution vulnerability",
+  "created_at": "2018-02-17",
+  "updated_at": "2018-04-09",
+  "publish_date": "2018-04-09",
+  "author": "Olivier Arteau (HoLyVieR)",
+  "module_name": "merge-objects",
+  "cves": [],
+  "vulnerable_versions": "",
```
Comment (lirantal) [inline vuln/npm/411.json]: here too

File: vuln/npm/407.json (hunk 4 above)
Comment (vdeturckheim) [reply vuln/npm/407.json]: My bad here; 1.0.1 is safe so it should get to patched_versions

File: vuln/npm/405.json [hunk 9]
```diff
@@ -0,0 +1,19 @@
+{
+  "id": 405,
+  "title": "Cross-site Scripting (XSS) - Stored",
+  "overview": "public allows to embed HTML in file names, which (in certain conditions) might lead to execute malicious JavaScript.",
+  "created_at": "2018-04-15",
```
Comment (lirantal) [inline vuln/npm/405.json]: all the created_at and updated_at dates through-out the vulns should be ISO 8601 with the recent contribution from @greysteil

File: vuln/npm/409.json (hunk 6 above)
Comment (vdeturckheim) [reply vuln/npm/409.json]: Should I get a free beer ? 🍻

**Comment by vdeturckheim** [comment]
@lirantal It should be good now :)

File: vuln/npm/405.json (hunk 9 above)
Comment (vdeturckheim) [reply vuln/npm/405.json]: I think YYYY-MM-DD is a valid ISO 8601 date 🤔 (see 5.2.1.1 in http://lists.ebxml.org/archives/ebxml-core/200104/pdf00005.pdf )

Comment (greysteil) [reply vuln/npm/405.json]: It is. I could look into adding a validation that minute-precision is specified, or something like that?

Comment (vdeturckheim) [reply vuln/npm/405.json]: I think precision smaller than day is a bit overkill for such records. What is your take here?

Comment (greysteil) [reply vuln/npm/405.json]: People might be watching the `updated_at` and want greater precision? I think I agree though - that's going to be pretty rare, and if they are then they'd be better of storing the commit SHA of the repo. Shall we change them all to dates? I'll put in a PR.

Comment (lirantal) [reply vuln/npm/405.json]: yeah it's valid as a partial, however all the vulns are formatted with the full timestamp

Comment (vdeturckheim) [reply vuln/npm/405.json]: Right SHAs should do the trick in times of need. Let's just see if @lirantal has an opinion here but I'd be 👍 for a PR.

File: vuln/npm/407.json (hunk 4 above)
Comment (lirantal) [reply vuln/npm/407.json]: 👍

File: vuln/npm/405.json (hunk 9 above)
Comment (lirantal) [reply vuln/npm/405.json]: I'm good 👍 
we can iterate on further enhancements as required

**Pull request merged**